### PR TITLE
Improve logging in a few areas

### PR DESCRIPTION
**Pull request by rowanc1**
image

### CODE REVIEW

File: apps/cli/src/logging/custom.ts [hunk 1]
```diff
@@ -34,7 +34,12 @@ export function getNpmLogger(session: ISession): LoggerDE {
     error(data: string) {
       const line = data.trim();
       if (!line) return;
-      if (line.includes('deprecated') || line === 'npm' || line.includes('WARN')) {
+      if (
+        line.includes('deprecated') ||
+        line.includes('package is no longer supported') ||
```
Comment (rowanc1) [inline apps/cli/src/logging/custom.ts:39]: This silences the crypto deprecation warning

File: apps/cli/src/cli/utils.ts [hunk 2]
```diff
@@ -159,7 +160,7 @@ export function clirun(
         useSession.log.debug(`\n\n${(error as Error)?.stack}\n\n`);
       }
       useSession.log.error((error as Error).message);
-      await logVersions(useSession, false);
+      logVersions(useSession, versions, false);
```
Comment (rowanc1) [inline apps/cli/src/cli/utils.ts:163]: logVersions is now not async, which was giving challenges in showing the error message in a different place than the process.exit when multiple things were running.

File: apps/cli/src/licenses/validators.ts [hunk 3]
```diff
@@ -84,12 +85,16 @@ export function validateLicense(input: any, opts: Options): License | undefined
...
   }
-  let value = validateString(input, opts);
-  if (value === undefined) return undefined;
-  value = value.toUpperCase();
+  const valueUnvalidated = validateString(input, opts);
+  if (valueUnvalidated === undefined) return undefined;
+  const value = valueUnvalidated.toUpperCase();
   if (!LICENSE_KEYS[value]) {
+    const possibleAlts = Object.keys(LICENSE_KEYS).filter((k) => k.includes(value));
+    const alts = possibleAlts.length > 0 ? ` Maybe try:\n- "${possibleAlts.join('"\n- "')}"\n` : '';
```
Comment (rowanc1) [inline apps/cli/src/licenses/validators.ts:93]: This was helpful for things like GPL, or CC-BY

File: apps/cli/src/toc/fromToc.ts [hunk 4]
```diff
@@ -51,7 +51,13 @@ export function projectFromToc(
   const pageSlugs: PageSlugs = {};
   const indexFile = resolveExtension(join(path, toc.root));
   if (!indexFile) {
-    throw Error(`Root from ${tocFile(path)} not found: ${indexFile}`);
+    throw Error(
```
Comment (rowanc1) [inline apps/cli/src/toc/fromToc.ts:54]: @stevejpurves this was the error hit by one of our users recently!

**Review by fwkoch — APPROVED** [review]

File: apps/cli/src/logging/custom.ts (hunk 1 above)
Comment (fwkoch) [reply apps/cli/src/logging/custom.ts:39]: Good to silence this for users. Should we resolve it at some point though? Not sure if there's another reason to stick with deprecated package.

Comment (rowanc1) [reply apps/cli/src/logging/custom.ts:39]: It is baked into some Jupyter dependencies, so yes, but might take a while for updates to happen.

**Pull request merged**